vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/grid/it/SortingPage.java

```diff
@@ -52,6 +57,80 @@ public SortingPage() {
         });
         button.setId("sort-by-age");
         add(button);
+
+        setSizeFull();
+        gui1 = new TestGui();
+        Button btnReset = new Button("reset", evt -> gui1.getGrid().sort(null));
+        btnReset.setId("btn-reset");
+        Button btRm = new Button("deattach", evt -> remove(gui1));
+        btRm.setId("btn-deattach");
+        Button btattach = new Button("attach",
+                evt -> add(gui1));
+        btattach.setId("btn-attach");
+        add(btRm, btattach,btnReset);
+    }
+
+    public class TestGui extends FlexLayout {
+        Grid<TestData> grid = new Grid<>(TestData.class);
+        public TestGui() {
+            setSizeFull();
+            grid.setSizeFull();
+            grid.setItems(createTestData());
+            grid.setId("second-grid");
+            // removes generated columns
+            grid.removeAllColumns();
```

Comment:
Don't create the grid with the bean class constructor if you want to define the columns manually.